source/game_sa/CBulletTraces.h

```diff
@@ -1,23 +1,22 @@
-/*
-    Plugin-SDK (Grand Theft Auto San Andreas) header file
-    Authors: GTA Community. See more here
-    https://github.com/DK22Pac/plugin-sdk
-    Do not delete this comment block. Respect others' work!
-*/
 #pragma once
-#include "PluginBase.h"
-#include "CBulletTrace.h"
-#include "CEntity.h"
+class CVector;
+#include <stdint.h>
 
-extern unsigned int MAX_NUM_BULLETTRACES; // default 16
+class CBulletTraces {
+public:
+
+    static CBulletTrace(&aTraces)[16];
 
-class  CBulletTraces {
+    static void InjectHooks();
+
+private:
 public:
-	static CBulletTrace *aTraces; // static CBulletTrace aTraces[16]
-	
-	static void AddTrace(CVector *start, CVector *end, int weaponType, CEntity *entity);
-	static void AddTrace(CVector *start, CVector *end, float radius, unsigned int time, unsigned char transparency);
-	static void Render();
-	static void Update();
-	static void Init();
+    static void Init();
+    static void AddTrace(const CVector& from, const CVector& to, float radius, uint32_t dissapearTime, uint8_t alpha);
+    static void Render();
+    static void AddTrace(const CVector& from, const CVector& to, eWeaponType weaponType, class CEntity* a4);
+    void Update();
+
+    static CBulletTrace* CBulletTraces::GetFree(); // Inlined function (Kinda NOTSA)
```

Comment:
Extra qualification on member 'GetFree'. Also will fail with ``/permissive-`` MSVC flag.

```suggestion
    static CBulletTrace* GetFree(); // Inlined function (Kinda NOTSA)
```

Reply:
Man I hate that I have to copy paste stuff from the source file into the header..
Is there some kind of tool which does it for me?
Because its the most annoying thing ever.

Reply:
You talking about source generator from IDB or what?

Reply:
I wrote this function myself, so I had to add it manually to both the header and source file, which is very annoying thing to do.